core_lib/src/interface/editor.cpp

```diff
@@ -751,7 +751,7 @@ bool Editor::importBitmapImage(QString filePath, int space)
         return false;
     }
 
-    while (reader.read(&img))
+    if (reader.read(&img))
```

Comment:
I see why you made this change, it freezes for me as well and I don't know why. Unfortunately this is an unacceptable change. From the Qt docs:
> For image formats that support animation, calling read() repeatedly will return the next frame. When all frames have been read, a null image will be returned.

Which means that animated gif export which also uses this function will only import the first frame. My recommended workaround here is to change this back to a while loop and then add this to the end of the loop :
```
// Workaround for tiff import getting stuck in this loop
if (!reader.supportsAnimation())
{
    break;
}
```

Reply:
I have a question. It says it will return a null image at the end of frames, not return a false.

Reply:
Very strange, must be a typo. The return type is definitely bool.

Reply:
Probably it's not a typo, there are 2 two overloaded QImageReader::read()
```
QImage QImageReader::read()
bool QImageReader::read(QImage *image)
```
and they shared the same function descriptions. So I think we could try to check whether the QImage is null, not the bool value.

Reply:
No that description was for `bool QImageReader::read(QImage *image)`, could be a copy and paste issue from the other method though. You could definitely try `while(!(img = reader.read()).isNull())`, but I'm not too optimistic that it will have any different results than the current approach.

Reply:
You're right. The result didn't change.